### PR TITLE
create_bezzier

### PR DESCRIPTION
**Pull request by prajwalprabhu**
pathced  MFEKmath::create_bazier::create

### CODE REVIEW

**Comment by ctrlcctrlv** [comment]
What does this do? Why is it named `CREATE` and not `SIMPLIFY`?

File: src/create_bezier.rs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    glif.outline = Some(final_result);
+    glifparser::write_to_filename(&glif, out_string).expect("Failed to write ");
+}
+pub fn clap_app() -> clap::App<'static> {
+    App::new("CREATE")
+    .setting(AppSettings::DeriveDisplayOrder)
+    .setting(AppSettings::AllowNegativeNumbers)
+    .about("Returns control points of an cubic bezier curve accorfing to knot(end) points")
+    .version("0.1.0")
+    .author("Fredrick Brennan <copypasteⒶkittens.ph>; Skia Authors; Andrew Hunter (flo_curves.rs); MFEK Authors")
```
Comment (ctrlcctrlv) [inline src/create_bezier.rs]: You can definitely put yourself as the author of parts you write :-)

**Comment by prajwalprabhu** [comment]
> What does this do? Why is it named `CREATE` and not `SIMPLIFY`?

It does not "simplify" in terms of fontforge's simplify  . And my plane is to use it for frontend.
where if user uses "create bezier" feature then user will give knot(end) point .But we have to calculate the control points. SO i named it "CREATE" .

**Comment by ctrlcctrlv** [comment]
I don't understand, please send some graphics

**Comment by prajwalprabhu** [comment]
> I don't understand, please send some graphics

In this curve the control points are determined by fontforge algorithm , by using "add a curve point" tool in fontforge.
close_curve
open_curve



If we want to add the feature to our front-end then we require an algorithm to generate control points according to knot(end) points.
So I implented that algorithm , and gave only end points from the above curve and algorithm generate control points , and the result is

close_curve_algo
open_curve_algo

**Comment by ctrlcctrlv** [comment]
I see, so this code guesses control point locations

**Comment by prajwalprabhu** [comment]
> I see, so this code guesses control point locations

Yes. But if the Contour<()> is mixture of curve and line ,the output won't be as expect.In that case we have to apply this algorithm part by part ,And I am implimenting it.

**Comment by ctrlcctrlv** [comment]
Okay, good. And I suppose that this will allow for curve fitting, if you know which segments are curved and which straight. You are doing well, that is the core of the problem.

**Comment by prajwalprabhu** [comment]
> Okay, good. And I suppose that this will allow for curve fitting, if you know which segments are curved and which straight. You are doing well, that is the core of the problem.
Ok,
So are you ok in merging the PR in MFEKMath.

**Comment by prajwalprabhu** [comment]
> Okay, good. And I suppose that this will allow for curve fitting, if you know which segments are curved and which straight. You are doing well, that is the core of the problem.

Are you merging the PR in MFEKMath or You will merge it after solving the combination part

**Comment by ctrlcctrlv** [comment]
The combination part based on point types is really important, I can't merge it without that I don't think.

**Comment by prajwalprabhu** [comment]
> The combination part based on point types is really important, I can't merge it without that I don't think.

Ya I have done it.Combination part is completed.

File: src/fit_to_points.rs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    glif.outline = Some(final_result);
+    glifparser::write_to_filename(&glif, out_string).expect("Failed to write ");
+}
+pub fn clap_app() -> clap::App<'static> {
+    App::new("FIT")
+    .setting(AppSettings::DeriveDisplayOrder)
+    .setting(AppSettings::AllowNegativeNumbers)
+    .about("Returns control points of an cubic bezier curve accorfing to knot(end) points")
+    .version("0.1.0")
+    .author("Fredrick Brennan <copypasteⒶkittens.ph>; Skia Authors; Andrew Hunter (flo_curves.rs); MFEK Authors")
```
Comment (ctrlcctrlv) [inline src/fit_to_points.rs]: Hello, please fix this, do not assign authorship to code that wasn't authored by that person; the only name here should be yours @prajwalprabhu

Comment (prajwalprabhu) [reply src/fit_to_points.rs]: Now is it Ok.

**Comment by ctrlcctrlv** [comment]
yes thanks :)

**Pull request merged**